MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -216,6 +216,27 @@ await using (cacheStream.ConfigureAwait(false))
                     contentType = MimeTypes.GetMimeType(new Uri(url).GetLeftPart(UriPartial.Path));
                 }
 
+                // Need to handle application/octet-stream as well as some providers, especially MENA providers use Blob storage like Azure
+                // and the content type is not set correctly on the blob properties. So you end up with the default application/octet-stream
+                // https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob?tabs=microsoft-entra-id#request-headers-all-blob-types
+                if (contentType.StartsWith("application/octet-stream", StringComparison.OrdinalIgnoreCase))
+                {
+                    var extension = Path.GetExtension(url).ToLowerInvariant();
```

Comment:
Can we use `MimeTypes.GetMimeType` here instead of creating a new mapping?

Reply:
@crobibero  The danger of stepping into a large code base wanting to bug squash without exploring first. Thanks for pointing out the method. Commit is in.